### DISCUSSION TITLE
Butler failed to send message to Teams with error code 500

### DISCUSSION THREAD

**Discussion by wieeem** [ptarmiganlabs/butler#724] [Q&A]
Hello,
I am currently involved in configuring Butler to send Teams notifications whenever a reload task fails on the QlikSense server.

 However, during this process, an error occurs with the following details: **Error: TEAMSNOTIF: Request failed with status code 500.**

This is the configuration for teamsNotification: 

![butler](https://github.com/ptarmiganlabs/butler/assets/47701160/d36c762b-5a83-4736-b613-62788ce52b03)

And this is the teams template that i've used : https://github.com/ptarmiganlabs/butler/blob/master/src/config/teams_templates/failed-reload.handlebars

This is the output error: 
![butler2](https://github.com/ptarmiganlabs/butler/assets/47701160/2d9f0fe7-ab4a-4503-84fc-514cdc51de4b)

- I have tested the webhook url of teams channel using curl request and it's working fine, so i don't think the problem is from my webhook teams.
- When i change type to **basic** the code error change to 400.
- I've also tested with two different butler version : 9.0.0 and 8.6.2 and i get always the same error.

I am now troubleshooting and looking for Help... Thanks in advance

**Comment by wieeem** [comment]
Hi again, 

- After several searches, I downloaded your source code and ran it locally and i had the same error msg.
- After debugging I noticed that in the header of the request sent in the request to teams the content-type is text, which should not be the case. according to the teams doc the content-type of the request should be **application/json.**
- This disucssion also allowed me to better understand the error: **https://learn.microsoft.com/en-us/answers/questions/1327800/(solved)-teams-incoming-webhook-bad-payload-receiv**

Suddenly what I did I modified the line **410** to "const res = await teamsWebhookObj.send(msg); " , in the function sendTeams , of the file https://github.com/ptarmiganlabs/butler/blob/master/src/lib/msteams_notification.js ,  which allows the request to automatically retrieve the type of data sent which is json . **and that solved the problem !**.

so can we consider this as a bug to fix in your code?

thanks in advance

**Comment by mountaindude** [comment]
Thx for reporting this and good that you got it working, but things should of course work out of the box too.

Recent versions of Butler include some updates to how Teams notifications are done.
Microsoft has changed some things in their APIs and webhooks, Butler is now updated to handle that properly.

Please try the latest version (as of today this is [9.1.2](https://github.com/ptarmiganlabs/butler/releases)) and see if that works for you too. From what I can see things now work fine.
